pyparsing/core.py

```diff
@@ -1408,7 +1412,8 @@ def __sub__(self, other) -> "ParserElement":
             raise TypeError(
                 f"Cannot combine element of type {type(other).__name__} with ParserElement"
             )
-        return self + And._ErrorStop() + other
+        # Construct And directly to ensure that an _ErrorStop is never the last element
```

Comment:
Interesting...

Reply:
It simplified the code in `And.__repr__()` a little to note that, in the only place where an `_ErrorStop` is created, it has another element after it.  If the user is creating an instance of an underscore-prefixed class, be it on their own head. :)

I guess it would still be possible to have a dangling `_ErrorStop` if `ParseExpression.exprs` is considered public API.  I can add a check for this if you'd like.